### PR TITLE
Issue #DIK-3929 fix: EPub not rendering properly on DIKSHA

### PR DESCRIPTION
**Pull request by rahulshukla**

### CODE REVIEW

File: player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js [hunk 1]
```diff
@@ -105,20 +100,35 @@ org.ekstep.contentrenderer.baseLauncher.extend({
...
     },
+
+    // Get the total number of actual pages to render
+    // remove page from pagination if in <spine> <itemref> property is linear=no
+    getTotalPages: function() {
+        var instance = this
+        var data = instance.book.locations.spine
+        var array = []
+        for (index = 0; index < data.length; index++) {
+          if (data[index].hasOwnProperty('linear') && data[index].linear != "no") {
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js:111]: String check always "toLowase"

File: player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js [hunk 2]
```diff
@@ -105,20 +100,35 @@ org.ekstep.contentrenderer.baseLauncher.extend({
...
+        return array.length;
+    },
+
     addEventHandlers: function () {
         var instance = this;
         instance.book.generatePagination().then(function (data) {
             instance._start = data[0].cfi;
-            instance.totalPages = data.length;
+            instance.totalPages = instance.getTotalPages();
+            if(instance.totalPages <= 1) instance.lastPage = true; // if all pages are non linear or only one page is linear
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js:123]: Why lessthan here. When instance.totalPages = 1 then instance.lastPage = true
When it is 0, we have to show message

File: player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js [hunk 3]
```diff
@@ -105,20 +100,35 @@ org.ekstep.contentrenderer.baseLauncher.extend({
         this.addEventHandlers();
         this.initProgressElements();
     },
+
+    // Get the total number of actual pages to render
+    // remove page from pagination if in <spine> <itemref> property is linear=no
+    getTotalPages: function() {
+        var instance = this
+        var data = instance.book.locations.spine
```
Comment (vinukumar-vs) [inline player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js:108]: Use _ library to any property is undefined. add this in try-catch block.

**Review by vinukumar-vs — CHANGES_REQUESTED** [review]

File: player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js (hunk 2 above)
Comment (rahulshukla) [reply player/public/coreplugins/org.ekstep.epubrenderer-1.0/renderer/plugin.js:123]: This is if spine has all the itemref as  **linear=no**, then our totalPages will be 0 and we always loads the first page then calculate the number of pages. This is corner case scenario. when total count page either 0 or 1

**Review by vinukumar-vs — APPROVED** [review]

**Comment by rjshrjndrn** [comment]
@vinukumar-vs Shall I merge this PR manually,  the new code doesn't have coverage.
cc @vrayulu

**Comment by vinukumar-vs** [comment]
@rjshrjndrn Yes, you can merge the PR. I have already approved it.

**Pull request merged**